components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/TextInputView.kt

```diff
@@ -39,139 +38,234 @@ import androidx.compose.ui.res.stringResource
-) {
-
-    var localValue: String by rememberSaveable { mutableStateOf("") }
-
-    var localError: String? by rememberSaveable { mutableStateOf(null) }
-
-    localError = errorText
-
-    val counterEnabled: Boolean = characterCount != null
-
-    fun errorText(): @Composable (() -> Unit)? {
-        return errorText?.let {
-            {
+object TextInputView {
+
+    private const val ERROR_TEXT_WITH_CHAR_COUNT_WIDTH = 0.8f
+    private const val CHAR_COUNT_WIDTH = 0.95f
+
+    @SuppressWarnings("LongMethod", "ComplexMethod")
+    @Composable
+    operator fun invoke(
+        modifier: Modifier = Modifier,
+        initialInputValue: String = "",
+        onInputValueChange: ((String) -> Unit)? = null,
+        inputFilter: ((String, String) -> String)? = null,
+        labelText: String? = null,
+        labelContentDescription: String? = null,
+        hintText: String? = null,
+        hintContentDescription: String? = null,
+        prefix: @Composable (() -> Unit)? = null,
+        placeholderText: String? = null,
+        errorText: String? = null,
+        errorContentDescription: String? = null,
+        characterCount: Int? = null,
+        singleLine: Boolean = false,
+        keyboardOptions: KeyboardOptions = KeyboardOptions.Default
+    ) {
+        var localValue: String by rememberSaveable { mutableStateOf(initialInputValue) }
+        var localError: String? by rememberSaveable { mutableStateOf(null) }
+        localError = errorText
+
+        val counterEnabled: Boolean = characterCount != null
+
+        val error: @Composable (() -> Unit)? = errorText?.let {
+            @Composable {
                 Text(
                     text = it,
-                    modifier = Modifier.semantics {
-                        if (errorContentDescription != null) {
-                            contentDescription = errorContentDescription
-                        }
-                    }
+                    modifier = Modifier.semantics { errorContentDescription?.let { contentDescription = it } }
                 )
             }
         }
-    }
 
-    @Composable
-    fun errorTextCounterCombo(): @Composable (() -> Unit) {
-        return {
+        val errorTextCounterCombo: @Composable (() -> Unit) = @Composable {
             Row {
                 Column(
                     horizontalAlignment = Alignment.Start,
-                    modifier = Modifier.fillMaxWidth(ErrorTextWithCharCountWidth)
+                    modifier = Modifier.fillMaxWidth(ERROR_TEXT_WITH_CHAR_COUNT_WIDTH)
                 ) {
                     if (errorText != null) {
                         Text(
                             text = errorText,
                             modifier = Modifier.semantics {
-                                if (errorContentDescription != null) {
-                                    contentDescription = errorContentDescription
+                                errorContentDescription?.let {
```

Comment:
Error text and the input field alignment are 
<img width="426" alt="Screenshot 2024-01-10 at 11 23 03" src="https://github.com/hmrc/android-components/assets/11981999/ea29a3f0-8b4b-46df-b6f3-25d7fb1bf333">
not matched with the design.

Reply:
I've updated the text style, but the padding field isn't exposed publicly yet so have checked with Aisling we can leave like this for now. 👍